### DISCUSSION TITLE
DS Port

### DISCUSSION THREAD

**Discussion by Cowlord2000** [diasurgical/DevilutionX#5363] [Ideas]
Is there any hope for a ds port I would love it, and tbh if I had the know how I would give it a shot. If it even is possible but I lack any knowledge what so ever to even port a game so I'm hoping some one else can do it, again if it even is possible.

**Comment by DakkJaniels** [comment]
Seems unlikely, it only has 4MB of RAM.

**Comment by AJenbo** [comment]
Currently the lower limit for running the game with out any reduction in quality looks to be about 32MB

To get it running on anything lower would require optimizing the assets, probably specifically for the target device, which would be a lot of work.

**Reply by julealgon** [reply]
Another option would be to reduce tile variation and animation frames (which the PS1 port did to some extent) which reduces the memory load without affecting actual behavior much (as long as the tile replacements are only visual ones... say, use a single spike tile in hell instead of many different types of spikes, etc).

Depending on how this is implemented, it could even scale automatically to a target RAM value, by loading more or less variations / frames dynamically.

**Reply by Cowlord2000** [reply]
Oh! So it's at least feasible?

**Reply by StephenCWills** [reply]
Climax Studios got the game to run on the PlayStation so.. yeah, technically feasible. I don't think it's likely to happen any time soon, if ever. And if it did happen, it would probably resemble the PlayStation version more than DevilutionX.

**Reply by Cowlord2000** [reply]
Hmm 🤔, well hopefully someone would like to give it ago on day but ig for now I'll just wait :o
